modules/javafx.graphics/src/main/java/com/sun/glass/ui/mac/MacApplication.java

```diff
@@ -73,14 +76,49 @@ native void _runLoop(ClassLoader classLoader, Runnable launchable,
                          boolean isTaskbarApplication);
     @Override
     protected void runLoop(final Runnable launchable) {
+        // For normal (not embedded) taskbar applications the masOS activation
+        // init code will deactivate and then reactivate the application to
+        // allow the system menubar to work properly.
+        // We need to spin up a nested event loop and wait for the reactivation
+        // to finish prior to allowing the rest of the initialization to run.
+        final Runnable wrappedRunnable = () -> {
+            if (isNormalTaskbarApp()) {
+                waitForReactivation();
+            }
+            launchable.run();
+        };
+
         isTaskbarApplication =
             AccessController.doPrivileged((PrivilegedAction<Boolean>) () -> {
                 String taskbarAppProp = System.getProperty("glass.taskbarApplication");
                 return  !"false".equalsIgnoreCase(taskbarAppProp);
             });
 
         ClassLoader classLoader = MacApplication.class.getClassLoader();
-        _runLoop(classLoader, launchable, isTaskbarApplication);
+        _runLoop(classLoader, wrappedRunnable, isTaskbarApplication);
+    }
+
+    private final CountDownLatch reactivationLatch = new CountDownLatch(1);
+
+    // Spin up a nested even loop waiting for the app reactivation event
```

Comment:
I just spotted a typo in this comment: even --> event

I'll wait to see if there are suggested changes from other reviewers, and then fix this.